instrumentation/opentelemetry-instrumentation-openai-v2/tests/test_chat_completions.py

```diff
@@ -102,56 +356,385 @@ def test_chat_completion_streaming(exporter, openai_client):
+    )
+
+    user_message = {
+        "content": "What's the weather in Seattle and San Francisco today?"
+    }
+    assert_message_in_logs(
+        logs[1], "gen_ai.user.message", user_message, spans[0]
+    )
+
+    choice_event = {
+        "index": 0,
+        "finish_reason": "tool_calls",
+        "message": {
+            "role": "assistant",
+            "tool_calls": [
+                {
+                    "id": tool_call_ids[0],
+                    "type": "function",
+                    "function": {
+                        "name": tool_names[0],
+                        "arguments": tool_args[0].replace("\n", ""),
+                    },
+                },
+                {
+                    "id": tool_call_ids[1],
+                    "type": "function",
+                    "function": {
+                        "name": tool_names[1],
+                        "arguments": tool_args[1].replace("\n", ""),
+                    },
+                },
+            ],
+        },
+    }
+    assert_message_in_logs(logs[2], "gen_ai.choice", choice_event, spans[0])
 
-    assert streaming_span.name == f"chat {llm_model_value}"
-    attributes = streaming_span.attributes
 
-    operation_name = attributes[GenAIAttributes.GEN_AI_OPERATION_NAME]
-    system = attributes[GenAIAttributes.GEN_AI_SYSTEM]
-    request_model = attributes[GenAIAttributes.GEN_AI_REQUEST_MODEL]
-    response_model = attributes[GenAIAttributes.GEN_AI_RESPONSE_MODEL]
-    response_id = attributes[GenAIAttributes.GEN_AI_RESPONSE_ID]
-    input_tokens = attributes[GenAIAttributes.GEN_AI_USAGE_INPUT_TOKENS]
-    output_tokens = attributes[GenAIAttributes.GEN_AI_USAGE_OUTPUT_TOKENS]
+def assert_message_in_logs(log, event_name, expected_content, parent_span):
+    assert log.log_record.attributes[EventAttributes.EVENT_NAME] == event_name
     assert (
-        operation_name == GenAIAttributes.GenAiOperationNameValues.CHAT.value
-    )
-    assert system == GenAIAttributes.GenAiSystemValues.OPENAI.value
-    assert request_model == llm_model_value
-    assert response_model == response_stream_model
-    assert response_id == response_stream_id
-    assert input_tokens == response_stream_usage.prompt_tokens
-    assert output_tokens == response_stream_usage.completion_tokens
-
-    events = streaming_span.events
-
-    # assert that the prompt and completion events are present
-    prompt_event = list(
-        filter(
-            lambda event: event.name == "gen_ai.content.prompt",
-            events,
-        )
+        log.log_record.attributes[GenAIAttributes.GEN_AI_SYSTEM]
+        == GenAIAttributes.GenAiSystemValues.OPENAI.value
     )
-    completion_event = list(
-        filter(
-            lambda event: event.name == "gen_ai.content.completion",
-            events,
-        )
+    assert json.dumps(log.log_record.body, sort_keys=True) == json.dumps(
+        expected_content, sort_keys=True
```

Comment:
```suggestion
    assert dict(log.log_record.body) == expected_content
```

Have you tried something like this? It's cheaper and the comparison is specialized so key order does not matter and you get the key that is different highlighted already

Reply:
thanks, will try!